Graphics resize fix

nessemis opened the pull request:
Graphics can now properly resize. Also looks better because rendering now happens on a per-pixel basis.

Known issues: UI doesn't update until the mouse hovers over one of it's elements. This will be fixed in a separate pull request (because the underlying issue isn't related to resizing).

nessemis commented on MeesGame/Gameplay/GameManagement/GameEnvironment.cs:
```diff
@@ -5,11 +5,13 @@
     protected InputHelper inputHelper;
     protected Matrix spriteScale;
-    protected Point windowSize;
+    protected Point preferedWindowSize;
```
I've renamed WindowSize to PreferedWindowSize because the window size isn't necessarily equal to this variable.

maxmeijer42 replied:
It's preferred not prefered

jensj12 commented:
The two PR's depend on each other, splitting them only makes merging harder (and they conflict each other).

jensj12 commented:
I don't like the fullscreen main menu, buttons are not properly centered in the bottom bar.

nessemis commented:
@jensj12 The two pull request solve completely separate issues (even though some issues are more apparent in this branch. I will change the positioning of the buttons right away though

nessemis commented:
@jensj12 The buttons are now organized left, center, right.

jensj12 commented:
I've asked @twjvdhorst to move the level editor button back to the main menu (and edit the strings so it fits). This works, but conflicts with that.
Another issue, the default screen size is much bigger than it used to be and already feels like fullscreen to me. Quitting fullscreen reposistions the window in the top-left corner of the game.

maxmeijer42 commented on MeesGame/Gameplay/GameManagement/GameEnvironment.cs:
```diff
@@ -29,19 +32,27 @@ public GameEnvironment()
     }
 
-    public Point WindowSize
+    public Point PreferedWindowSize
```
Spelling

jensj12 commented:
Please fix all conflicts ASAP. This branch fixes some issues introduced in #208.

nessemis commented:
@jensj12  fixed

jensj12 commented:
Please do your merges properly. I don't like seeing changes from other branches.

jensj12 reviewed (changes requested):
The level editor button is gone.

Commit:
four buttons with equal distance

nessemis commented:
@jensj12  @maxmeijer42 I've fixed both issues. I don't know what went wrong with merging, but I fixed it by hand-picking the needed changes. (the scrollbars should be fixed now as well, somehow got into the wrong branch).

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into graphics_resize_fix

# Conflicts:
#	MeesGame/Gameplay/GameStates/LevelEditorState.cs

maxmeijer42 commented:
@nessemis No, it's not fixed. There were two instances.

jensj12 commented:
What is going on here? It seems the changes have been moved to a new branch 'new_resize_fix' and this branch is left with only the extra buttons fix. @nessemis don't mess up now, we need this change.
